dotnet/src/Planners/Planners.Handlebars.UnitTests/Handlebars/HandlebarsPlannerTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.ChatCompletion;
+using Microsoft.SemanticKernel.Planning.Handlebars;
+using Moq;
+using Xunit;
+
+namespace Microsoft.SemanticKernel.Planners.UnitTests.Handlebars;
+
+public sealed class HandlebarsPlannerTests
+{
+    private const string PlanString =
+    @"```handlebars
+        <plan>
+            <function.SummarizePlugin.Summarize/>
+            <function.WriterPlugin.Translate language=""French"" setContextVariable=""TRANSLATED_SUMMARY""/>
+            <function.email.GetEmailAddress input=""John Doe"" setContextVariable=""EMAIL_ADDRESS""/>
+            <function.email.SendEmail input=""$TRANSLATED_SUMMARY"" email_address=""$EMAIL_ADDRESS""/>
+        </plan>```";
+
+    [Theory]
+    [InlineData("Summarize this text, translate it to French and send it to John Doe.")]
+    public async Task ItCanCreatePlanAsync(string goal)
+    {
+        // Arrange
+        var plugins = this.CreatePluginCollection();
+        var kernel = this.CreateKernel(PlanString, plugins);
+        var planner = new HandlebarsPlanner();
+
+        // Act
+        HandlebarsPlan plan = await planner.CreatePlanAsync(kernel, goal);
+
+        // Assert
+        Assert.True(!string.IsNullOrEmpty(plan.Prompt));
+        Assert.True(!string.IsNullOrEmpty(plan.ToString()));
+    }
+
+    [Fact]
+    public async Task EmptyGoalThrowsAsync()
+    {
+        // Arrange
+        var kernel = this.CreateKernel(PlanString);
+
+        var planner = new HandlebarsPlanner();
+
+        // Act & Assert
+        await Assert.ThrowsAsync<ArgumentException>(async () => await planner.CreatePlanAsync(kernel, string.Empty));
+    }
+
+    [Fact]
+    public async Task InvalidXMLThrowsAsync()
+    {
+        // Arrange
+        var kernel = this.CreateKernel("<plan>notvalid<</plan>");
+
+        var planner = new HandlebarsPlanner();
+
+        // Act & Assert
+        var exception = await Assert.ThrowsAsync<KernelException>(async () => await planner.CreatePlanAsync(kernel, "goal"));
+        Assert.True(exception?.Message?.Contains("Could not find the plan in the results", StringComparison.InvariantCulture));
+    }
+
+    private Kernel CreateKernel(string testPlanString, KernelPluginCollection? plugins = null)
```

Comment:
nit so it's more indicative of which dependencies are in our control vs variable

```suggestion
    private Kernel CreateKernelWithMockCompletionResult(string testPlanString, KernelPluginCollection? plugins = null)
```

Reply:
Taken